File: apple/Sources/CCDAEngine/Models/CCDAHeader.swift
import Foundation

public struct CCDAHeader {
    public let realmCode: String?
    public let typeId: CCDAIdentifier?
    public let templateIds: [CCDATemplateId]
    public let documentId: CCDAIdentifier
    public let code: CCDACodedValue?
    public let title: String?
    public let effectiveTime: String?
    public let confidentialityCode: CCDACodedValue?
    public let languageCode: String?
}

File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Document/CCDAHeader.kt
package com.ccdakit.engine

/**
 * Header metadata parsed from the root C-CDA document.
 *
 * @property realmCode Realm code such as `US`.
 * @property typeId CDA type identifier.
 * @property templateIds Document-level template identifiers.
 * @property documentId Document identifier.
 * @property code Document type code.
 * @property title Document title.
 * @property effectiveTime Document effective timestamp.
 * @property confidentialityCode Confidentiality code.
 * @property languageCode Document language code.
 */
data class CCDAHeader(
    val realmCode: String? = null,
    val typeId: CCDAIdentifier? = null,
    val templateIds: List<CCDATemplateId> = emptyList(),
    val documentId: CCDAIdentifier,
    val code: CCDACodedValue? = null,
    val title: String? = null,
    val effectiveTime: CCDATimestamp? = null,
    val confidentialityCode: CCDACodedValue? = null,
    val languageCode: String? = null,
)
